mflow_processor/bsread_writer.py

```diff
@@ -128,27 +158,48 @@ def start(self):
 
         self._running_event.clear()
 
-        start_event = Event()
-        start_event.set()
-
-        self._receiving_thread = Thread(target=self.receive_messages, args=(self._running_event,
-                                                                            address,
-                                                                            self.output_file,
-                                                                            self.receive_timeout,
-                                                                            self.n_pulses,
-                                                                            start_event))
-
+        self._receiving_thread = Thread(target=self.receive_messages, args=(address, ))
         self._receiving_thread.start()
 
         if not self._running_event.wait(BSREAD_START_TIMEOUT):
-            start_event.clear()
             raise ValueError("Cannot start bsread writing process in time.")
 
     def stop(self):
-        self._logger.debug("Writer stopped.")
-
+        self._logger.debug("Stopping bsread_writer.")
         self._running_event.clear()
 
         if self._receiving_thread:
             self._receiving_thread.join()
+            self._logger.debug("Join bsread receiving thread.")
+
+        if self._writing_thread:
+            self._writing_thread.join()
+            self._logger.debug("Join bsread writing thread.")
 
+        self._logger.debug("bsread_writer stopped.")
+
+    @staticmethod
+    def write_message_to_hdf5(h5_file, message, first_iteration):
+        """ Placeholder for a function to write message's content into hdf5 file.
+
+        Intended for debugging. Saves only pulse_ids of messages!
+        """
+        if first_iteration:
+            dset_pulse_id = h5_file.create_dataset('pulse_id', dtype='i8', shape=(1, 1), maxshape=(None, 1))
+        else:
+            dset_pulse_id = h5_file['pulse_id']
+            dset_pulse_id.resize(dset_pulse_id.shape[0] + 1, axis=0)
+
+        dset_pulse_id[-1] = message.data.pulse_id
+
+    @staticmethod
+    def prune_messages_in_hdf5(h5_file, end_pulse_id):
+        """ Placeholder for a function to prune hdf5 file content down to end_pulse_id.
+
+        Intended for debugging.
+        """
+        dset_pulse_id = h5_file['pulse_id']
+        while dset_pulse_id[-1] > end_pulse_id:
+            # this will also discard the data
+            # see the Note at http://docs.h5py.org/en/latest/high/dataset.html#resizable-datasets
+            dset_pulse_id.resize(dset_pulse_id.shape[0] - 1, axis=0)
```

Comment:
There is too much resizing going on here. Just find the last element you want to keep, and then resize the dataset only once.